prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelInitializeCommand.java

```diff
@@ -0,0 +1,175 @@
+import org.wildfly.prospero.model.ChannelRef;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+@CommandLine.Command(
+        name = CliConstants.Commands.CUSTOMIZATION_INITIALIZE_CHANNEL,
+        aliases = {CliConstants.Commands.CUSTOMIZATION_INIT_CHANNEL},
+        sortOptions = false
+)
+public class ChannelInitializeCommand extends AbstractCommand {
+    public static final String CUSTOMIZATION_REPO_ID = "customization-repository";
+    public static final String CUSTOM_CHANNELS_GROUP_ID = "custom.channels";
+    public static final String DEFAULT_CUSTOMIZATION_REPOSITORY = "customization-repository";
+    @CommandLine.Option(
+            names = CliConstants.CUSTOMIZATION_CHANNEL_NAME
+    )
+    private Optional<String> name;
+
+    @CommandLine.Option(
+            names = CliConstants.CUSTOMIZATION_REPOSITORY_URL,
+            descriptionKey = "customization-repository"
+    )
+    private Optional<URL> repositoryUrl;
+
+    @CommandLine.Option(names = CliConstants.DIR)
+    Optional<Path> directory;
+
+
+    public ChannelInitializeCommand(Console console, ActionFactory actionFactory) {
+        super(console, actionFactory);
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        Path installationDirectory = determineInstallationDirectory(directory);
+        MetadataAction metadataAction = actionFactory.metadataActions(installationDirectory);
+
+        if (!validateRepository(metadataAction) || !validateChannel()) {
+            return ReturnCodes.INVALID_ARGUMENTS;
+        }
+
+        final URL url;
+        if (repositoryUrl.isPresent()) {
+            url = repositoryUrl.get();
+            if (repositoryUrl.get().getProtocol().equals("file")) {
+                if (!createLocalRepository(Paths.get(repositoryUrl.get().toURI()))) {
+                    return ReturnCodes.PROCESSING_ERROR;
+                }
+            }
+        } else {
+            final Path defaultLocalRepoPath = installationDirectory.resolve(InstallationMetadata.METADATA_DIR)
+                    .resolve(DEFAULT_CUSTOMIZATION_REPOSITORY);
+            if (!createLocalRepository(defaultLocalRepoPath)) {
+                return ReturnCodes.PROCESSING_ERROR;
+            }
+            url = defaultLocalRepoPath.toUri().toURL();
+        }
+
+        if (name.isEmpty()) {
+            if (metadataAction.getChannels().stream().anyMatch(c->c.getGav()!=null && c.getGav().startsWith(CUSTOM_CHANNELS_GROUP_ID + ":"))) {
+                console.error(CliMessages.MESSAGES.customizationChannelAlreadyExists());
+                return ReturnCodes.PROCESSING_ERROR;
+            }
+            name = Optional.of(CUSTOM_CHANNELS_GROUP_ID + ":" + RandomStringUtils.randomAlphanumeric(8));
```

Comment:
I'm not sure why there is the random part, as it seems that only single `CUSTOM_CHANNELS_GROUP_ID + ":"` channel is allowed.

Reply:
This is in case the repository is used to host multiple channels. Each server can be subscribed to a different channel

Reply:
OK, I think I finally got it. The channel yaml is going to be published in the customization repo, under this random name, to avoid conflicts with other installations that also have customization channel.

Why is there only single customization channel allowed in an installation?